lib/api/src/entities/engine/inner.rs

```diff
@@ -40,6 +40,140 @@ impl BackendEngine {
+                Features::default()
+            }
+
+            // For other backends, provide hardcoded defaults
+            _ => Features::default(),
+        }
+    }
+
+    /// Returns all WebAssembly features that this backend is capable of supporting for the given target.
+    #[inline]
+    pub fn supported_features_for_target(backend: &BackendKind, target: &Target) -> Features {
+        // For backends, use the appropriate supported features
+        match backend {
+            #[cfg(feature = "sys")]
+            crate::BackendKind::Sys => {
+                // For SYS backend, get compiler-specific features if available
+                #[cfg(feature = "cranelift")]
+                {
+                    let cranelift_config = wasmer_compiler_cranelift::Cranelift::new();
+                    let mut features = cranelift_config.supported_features_for_target(target);
+                    features.exceptions(false); // Cranelift doesn't support exceptions
+                    return features;
+                }
+
+                #[cfg(all(feature = "llvm", not(feature = "cranelift")))]
+                {
+                    let llvm_config = wasmer_compiler_llvm::LLVM::new();
+                    let mut features = llvm_config.supported_features_for_target(target);
+                    features.exceptions(true); // LLVM supports exceptions
+                    return features;
+                }
+
+                #[cfg(all(
+                    feature = "singlepass",
+                    not(feature = "cranelift"),
+                    not(feature = "llvm")
+                ))]
+                {
+                    let singlepass_config = wasmer_compiler_singlepass::Singlepass::new();
+                    let mut features = singlepass_config.supported_features_for_target(target);
+                    features.multi_value(false); // Singlepass doesn't support multi-value
+                    features.exceptions(false); // Singlepass doesn't support exceptions
+                    return features;
+                }
+
+                // Default SYS features if no compiler is enabled
+                let mut features = Features::default();
+                features.bulk_memory(true);
+                features.reference_types(true);
+                features.simd(true);
+                features.threads(true);
+                features.multi_value(true);
+                features.exceptions(false);
+                features
+            }
+
+            #[cfg(feature = "v8")]
+            crate::BackendKind::V8 => {
+                // V8-specific features
+                let mut features = Features::default();
+                features.bulk_memory(true);
+                features.reference_types(true);
+                features.simd(true);
+                features.threads(true);
+                features.multi_value(true);
+                features.exceptions(false);
+                features
```

Comment:
This needs to be implemented on `v8::Engine::supported_features_for_target(target)`